pilot/pkg/serviceregistry/kube/controller/ambient/workloads.go

```diff
@@ -542,3 +557,59 @@ func implicitWaypointPolicies(ctx krt.HandlerContext, Waypoints krt.Collection[W
 		return ptr.Of(w.Namespace + "/" + policy)
 	})
 }
+
+func gatewayUID(gw model.NetworkGateway) string {
+	return "NetworkGateway/" + string(gw.Network) + "/" + gw.Addr + "/" + strconv.Itoa(int(gw.HBONEPort))
```

Comment:
Hmm, I am not sure I understand. Ztunnel is not looking up a specific UID, it is subscribing to all resources.

The only time it does lookup specific is with on-demand, where it looks up by IP address. And since the workload has an IP it would be returned just like any other